Modules/Filtering/Thresholding/include/itkIntermodesThresholdImageFilter.hxx

```diff
@@ -0,0 +1,159 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkIntermodesThresholdImageFilter_hxx
+#define itkIntermodesThresholdImageFilter_hxx
+#include "itkIntermodesThresholdImageFilter.h"
+
+#include "itkImageToHistogramFilter.h"
+#include "itkMaskedImageToHistogramFilter.h"
+#include "itkBinaryThresholdImageFilter.h"
+#include "itkMaskImageFilter.h"
+#include "itkProgressAccumulator.h"
+
+namespace itk
+{
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+IntermodesThresholdImageFilter<TInputImage, TOutputImage, TMaskImage>::IntermodesThresholdImageFilter()
+{
+  m_Calculator = CalculatorType::New();
+  this->SetCalculator(m_Calculator);
+  m_Calculator->SetMaximumSmoothingIterations(10000);
+  m_Calculator->SetUseInterMode(true);
+}
+
+template <typename TInputImage, typename TOutputImage, typename TMaskImage>
+void
+IntermodesThresholdImageFilter<TInputImage, TOutputImage, TMaskImage>::SetMaximumSmoothingIterations(
+  SizeValueType maxSmoothingIterations)
+{
+  m_Calculator->SetMaximumSmoothingIterations(maxSmoothingIterations);
```

Comment:
Does this cause the modify time of the image filter class to be updated?

The simple way to do this would be to copy the code in the set/get macro which check if the value is the same, if they are different then set the modify time. The more complicated way would involve modify the parent's GetMTime method to return the max of the MTime of itself, and the m_Calculator.

Reply:
Opened issue #1720.